NextcloudTalk/SettingsTableViewController.swift

```diff
@@ -0,0 +1,676 @@
+            let activeAccount = NCDatabaseManager.sharedInstance().activeAccount()
+            if activeAccount.phone.isEmpty {
+                self.presentSetPhoneNumberDialog()
+            }
+        }
+    }
+
+    func presentSetPhoneNumberDialog() {
+        let alertTitle = NSLocalizedString("Phone number", comment: "")
+        let alertMessage = NSLocalizedString("You can set your phone number so other users will be able to find you", comment: "")
+        let setPhoneNumberDialog = UIAlertController(title: alertTitle, message: alertMessage, preferredStyle: .alert)
+
+        setPhoneNumberDialog.addTextField { [self] textField in
+            let location = NSLocale.current.regionCode
+            textField.text = "+\(self.phoneUtil?.getCountryCode(forRegion: location) ?? 0)"
+            do {
+                let exampleNumber: NBPhoneNumber =  try phoneUtil!.getExampleNumber(location)
+                textField.placeholder = try phoneUtil?.format(exampleNumber, numberFormat: NBEPhoneNumberFormat.INTERNATIONAL)
+            } catch {
+                print("Error with phone number format...")
+            }
+            textField.keyboardType = .phonePad
+            textField.delegate = self
+            textField.tag = kPhoneTextFieldTag
+        }
+        setPhoneAction = UIAlertAction(title: NSLocalizedString("Set", comment: ""), style: .default, handler: { _ in
+            let phoneNumber = setPhoneNumberDialog.textFields?[0].text
+
+            NCAPIController.sharedInstance().setUserProfileField(kUserProfilePhone, withValue: phoneNumber, for: NCDatabaseManager.sharedInstance().activeAccount()) { error, _ in
+                if error != nil {
+                    self.presentPhoneNumberErrorDialog(phoneNumber: phoneNumber!)
+                    print("Error setting phone number ", error ?? "")
+                } else {
+                    let alert = UIAlertController(title: NSLocalizedString("Phone number set successfully", comment: ""), message: "", preferredStyle: .alert)
+                    alert.addAction(UIAlertAction(title: "OK", style: .default, handler: nil))
+                }
+                self.refreshUserProfile()
+            }
+        })
+
+        setPhoneAction!.isEnabled = false
+        setPhoneNumberDialog.addAction(setPhoneAction!)
+
+        let cancelAction = UIAlertAction(title: NSLocalizedString("Skip", comment: ""), style: .default) { _ in
+            self.refreshUserProfile()
+        }
+        setPhoneNumberDialog.addAction(cancelAction)
+        self.present(setPhoneNumberDialog, animated: true, completion: nil)
+    }
+
+    func presentPhoneNumberErrorDialog(phoneNumber: String) {
+        var failedPhoneNumberDialog: UIAlertController?
+        var failedPhoneNumber: NBPhoneNumber?
+        do {
+            failedPhoneNumber = try phoneUtil!.parse(phoneNumber, defaultRegion: nil)
+        } catch {
+            print("Error...")
+        }
```

Comment:
Do we really need a try-catch block here?
If yes, please add a more descriptive error message.